src/lib.rs

```diff
@@ -1,14 +1,48 @@
 use std::fmt;
 
-/// TODO...
+/// A trait that can be used for converting between different color models
+/// and performing various transformations on them.
 pub trait Color {
-    // TODO...
+    /// Constructs a valid CSS color string for a given color type.
+    ///
+    /// # Examples
+    /// ```
+    /// use css_colors::{Color, RGB, RGBA};
+    ///
+    /// let salmon = RGB { r: 250, g: 128, b: 114 };
+    /// let opaque_salmon = RGBA { r: 250, g: 128, b: 114, a: 0.5 };
+    ///
+    /// assert_eq!(salmon.to_css(), "rgb(250, 128, 114)");
+    /// assert_eq!(opaque_salmon.to_css(), "rgba(250, 128, 114, 0.5)");
+    /// ```
     fn to_css(self) -> String;
 
-    // TODO...
+    /// Converts self into its RGB representation.
+    /// When converting from a color model that supports an alpha channel
+    /// (e.g. RGBA), the alpha value will not be preserved.
+    ///
+    /// # Examples
+    /// ```
+    /// use css_colors::{Color, RGB, RGBA};
+    ///
+    /// let opaque_tomato = RGBA { r: 255, g: 99, b: 71, a: 0.5 };
+    ///
+    /// assert_eq!(opaque_tomato.to_rgb(), RGB { r: 255, g: 99, b: 71 });
+    /// ```
     fn to_rgb(self) -> RGB;
 
-    // TODO...
+    /// Converts self into its RGBA representation.
+    /// When converting from a color model that does not supports an alpha channel
+    /// (e.g. RGB), it will be treated as fully opaque (i.e. with a value of 1.0).
```

Comment:
Maybe we should just ✂️ "(i.e. with a value of 1.0)" to be consistent with your other PR